homeassistant/components/epion/sensor.py

```diff
@@ -0,0 +1,137 @@
+        name="Temperature",
+        suggested_display_precision=1,
+    ),
+    SensorEntityDescription(
+        device_class=SensorDeviceClass.HUMIDITY,
+        native_unit_of_measurement=PERCENTAGE,
+        key="humidity",
+        name="Humidity",
+        suggested_display_precision=1,
+    ),
+    SensorEntityDescription(
+        device_class=SensorDeviceClass.ATMOSPHERIC_PRESSURE,
+        native_unit_of_measurement=UnitOfPressure.HPA,
+        key="pressure",
+        name="Pressure",
+        suggested_display_precision=0,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add an Epion entry."""
+    coordinator: EpionCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    entities = [
+        EpionSensor(coordinator, epion_device, description)
+        for epion_device in coordinator.data.values()
+        for description in SENSOR_TYPES
+    ]
+
+    async_add_entities(entities)
+
+
+class EpionSensor(CoordinatorEntity, SensorEntity):
+    """Representation of an Epion Air sensor."""
+
+    def __init__(
+        self,
+        coordinator: EpionCoordinator,
+        epion_device: dict,
+        description: SensorEntityDescription,
+    ) -> None:
+        """Initialize an EpionSensor."""
+        super().__init__(coordinator)
+        self._epion_coordinator = coordinator
+        self._epion_device = epion_device
+        self._measurement_key = description.key
+        self._display_name = description.name
+        self._attr_native_unit_of_measurement = description.native_unit_of_measurement
+        self._attr_device_class = description.device_class
+        self._attr_suggested_display_precision = description.suggested_display_precision
+        self._attr_state_class = SensorStateClass.MEASUREMENT
+        self.unique_id = f"{self._epion_device['deviceId']}_{self._measurement_key}"
+        self.has_entity_name = True
+        self.name = description.name
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the value reported by the sensor."""
+        return self.extract_value()
+
+    @property
+    def available(self) -> bool:
+        """Return the availability of this sensor."""
+        return self.extract_value() is not None
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return the device info."""
+        my_device_id = self._epion_device["deviceId"]
+        sw_version: str | None = None
+        if my_device_id in self._epion_coordinator.data:
+            current_device = self._epion_coordinator.data[my_device_id]
+            sw_version = current_device.get("fwVersion")
+
+        return DeviceInfo(
+            identifiers={(DOMAIN, self._epion_device["deviceId"])},
+            manufacturer="Epion",
+            name=self._epion_device["deviceName"],
+            sw_version=sw_version,
+            model="Epion Air",
+        )
```

Comment:
With the new `device` property, this code is simple and can be moved to the constructor with `_attr_device_info = ...`